source/blog/2017-10-16-fall-in-for-finagle.md

```diff
@@ -0,0 +1,144 @@
+
+#### Scrooge Updates
+
+We've been making lots of changes to improve the usability and performance of
+Scrooge and Scrooge-generated code. This includes a set of refactorings to modernize the generated code so that it’s written in terms of Filters and Services!
+
+This will allow us to address some long standing issues like server-side response classification and correctly stating on serialization/deserialization failures.
+
+Additionally, we going to try to update to libthrift 0.10.0 from 0.5.0. This is still exploratory and we’re working to understand the ramifications.
+
+#### Bringing Backup Requests Back
+
+Finagle has had a
+[BackupRequestFilter](https://github.com/twitter/finagle/blob/develop/finagle-exp/src/main/scala/com/twitter/finagle/exp/BackupRequestFilter.scala)
+since [late
+2012](https://github.com/twitter/finagle/commit/526e508579309711a9c56007eff2a783826d331c)
+and evolved into its current form in [early
+2013](https://github.com/twitter/finagle/commit/8ca92593aac1d3f5bd4741d6b8e6fcf26053dc44).
+The filter has a variety of what we would consider flaws (too many
+configuration knobs, less than ideal memory characteristics in many cases,
+lack of integration into the Stack-APIs and ClientBuilder to name a few) which
+has relegated it to Finagle's experimental module.
+
+We're investigating what it would take to give users a single configuration
+variable -- ideally the percentage of additional requests to send -- to
+configure sending a backup request to another server.
+
+#### ‘Push’ based API for Protocols
+
+Modern multiplexing protocols are characteristically very stateful and event
+driven making them challenging to implement with the current ‘pull’ based
+abstractions available in Finagle.
+
+We are currently working towards the introduction of a Finagle ‘push’ based
+model which when coupled with a single threaded concurrency model would
+provide a more efficient and natural foundation for implementing multiplexing
+protocols.
+
+#### Application-Level Streaming Research
+
+Related to the above, Finagle has historically been dependent on
+request-response style protocols: a single request would be sent by the client
+and the server would reply with a single response. And then later multiplexing
+protocols to alleviate head-of-line blocking. However, all the protocols in
+Finagle are expressed as request-response APIs with some limited support for
+streaming through using Reader and AsyncStream
+[[1](https://github.com/twitter/finagle/blob/develop/finagle-example/src/main/scala/com/twitter/finagle/example/http/HttpStreamingServer.scala)].
+These are somewhat low-level constructs and can be hard to use properly.
+
+We are investigating the design of an API for bidirectional streaming of objects not just bytes. Stay tuned.
+
+#### Blocking Code Detection
+
+Our newest intern [@andrewmccree14](https://twitter.com/andrewmccree14) is
+working on a project to help detect code that is blocking in Finagle. This
```

Comment:
s/code that is blocking/blocking code

Reply:
That's just repeating the section title!

Reply:
Ha, fair enough :)